File: frontend/src/types/analysis.ts
export enum AnalysisStatus {
  PENDING = 'pending',
  PROCESSING = 'processing',
  COMPLETED = 'completed',
  FAILED = 'failed',
  CANCELLED = 'cancelled'
}

export interface AnalysisParameters {
  market_type: 'A股'
  analysis_date?: string
  research_depth: '快速' | '基础' | '标准' | '深度' | '全面' | string
  selected_analysts: string[]
  custom_prompt?: string
  include_sentiment?: boolean
  include_risk?: boolean
  include_charts?: boolean
  language: 'zh-CN'
  quick_analysis_model?: string
  deep_analysis_model?: string
}

export interface AnalysisResult {
  analysis_id?: string
  summary?: string
  recommendation?: string
  confidence_score?: number
  risk_level?: string
  key_points: string[]
  detailed_analysis?: Record<string, any>
  reports?: Record<string, string>
  charts: string[]
  tokens_used: number
  execution_time: number
  error_message?: string
}

export interface AnalysisTask {
  id?: string
  task_id: string
  user_id?: string
  symbol?: string
  stock_code?: string
  stock_name?: string
  status: AnalysisStatus
  progress: number
  created_at: string
  started_at?: string
  completed_at?: string
  worker_id?: string
  parameters: AnalysisParameters
  result?: AnalysisResult
  retry_count?: number
  max_retries?: number
  last_error?: string
}

export interface StockInfo {
  symbol: string
  code?: string
  full_symbol?: string
  name: string
  market: 'A股' | string
  industry?: string
  area?: string
  board?: string
  exchange?: string
  total_mv?: number
  circ_mv?: number
  pe?: number
  pb?: number
  pe_ttm?: number
  pb_mrq?: number
  roe?: number
  close?: number
  pct_chg?: number
  amount?: number
  turnover_rate?: number
  volume_ratio?: number
  ma20?: number
  rsi14?: number
  kdj_k?: number
  kdj_d?: number
  kdj_j?: number
  dif?: number
  dea?: number
  macd_hist?: number
}

export interface SingleAnalysisRequest {
  symbol?: string
  stock_code?: string
  parameters?: AnalysisParameters
}

export interface AnalysisTaskResponse {
  task_id: string
  symbol?: string
  stock_code?: string
  stock_name?: string
  status: AnalysisStatus
  progress: number
  created_at: string
  started_at?: string
  completed_at?: string
  result?: AnalysisResult
}

export interface AnalysisHistoryQuery {
  status?: AnalysisStatus
  start_date?: string
  end_date?: string
  symbol?: string
  stock_code?: string
  page: number
  page_size: number
}

export interface AnalysisHistoryResponse {
  tasks: AnalysisTask[]
  total: number
  page: number
  page_size: number
  has_more: boolean
}

export interface TaskProgress {
  task_id: string
  status: AnalysisStatus
  progress: number
  message?: string
  updated_at: string
}

export interface AnalysisStats {
  total_analyses: number
  successful_analyses: number
  failed_analyses: number
  success_rate: number
  average_execution_time: number
  total_tokens_used: number
  daily_analyses: number
  monthly_analyses: number
}

export interface QueueStatus {
  pending: number
  processing: number
  completed: number
  failed: number
  queue_size: number
}

export interface UserQueueStatus {
  pending: number
  processing: number
  concurrent_limit: number
  available_slots: number
}

export interface AnalysisReport {
  id: string
  task_id?: string
  title: string
  content: string
  format: 'html' | 'pdf' | 'markdown' | 'json'
  created_at: string
  download_url?: string
}

File: app/models/analysis.py
from datetime import datetime
from enum import Enum
from typing import Any, Dict, List, Optional

from pydantic import BaseModel, ConfigDict, Field, field_serializer

from .user import PyObjectId
from app.utils.timezone import now_tz


class AnalysisStatus(str, Enum):
    PENDING = "pending"
    PROCESSING = "processing"
    COMPLETED = "completed"
    FAILED = "failed"
    CANCELLED = "cancelled"


class AnalysisParameters(BaseModel):
    market_type: str = "A股"
    analysis_date: Optional[datetime] = None
    research_depth: str = "标准"
    selected_analysts: List[str] = Field(default_factory=lambda: ["market", "fundamentals", "news", "social"])
    custom_prompt: Optional[str] = None
    include_sentiment: bool = True
    include_risk: bool = True
    language: str = "zh-CN"
    quick_analysis_model: Optional[str] = "qwen-turbo"
    deep_analysis_model: Optional[str] = "qwen-max"


class AnalysisResult(BaseModel):
    analysis_id: Optional[str] = None
    summary: Optional[str] = None
    recommendation: Optional[str] = None
    confidence_score: Optional[float] = None
    risk_level: Optional[str] = None
    key_points: List[str] = Field(default_factory=list)
    detailed_analysis: Optional[Dict[str, Any]] = None
    reports: Dict[str, Any] = Field(default_factory=dict)
    charts: List[str] = Field(default_factory=list)
    tokens_used: int = 0
    execution_time: float = 0.0
    error_message: Optional[str] = None
    model_info: Optional[str] = None


class AnalysisTask(BaseModel):
    id: Optional[PyObjectId] = Field(default_factory=PyObjectId, alias="_id")
    task_id: str = Field(..., description="任务唯一标识")
    user_id: PyObjectId
    symbol: str = Field(..., description="6位A股代码")
    stock_code: Optional[str] = Field(None, description="兼容字段，优先使用symbol")
    stock_name: Optional[str] = None
    status: AnalysisStatus = AnalysisStatus.PENDING
    progress: int = Field(default=0, ge=0, le=100)
    created_at: datetime = Field(default_factory=now_tz)
    started_at: Optional[datetime] = None
    completed_at: Optional[datetime] = None
    worker_id: Optional[str] = None
    parameters: AnalysisParameters = Field(default_factory=AnalysisParameters)
    result: Optional[AnalysisResult] = None
    retry_count: int = 0
    max_retries: int = 3
    last_error: Optional[str] = None

    model_config = ConfigDict(populate_by_name=True, arbitrary_types_allowed=True)


class StockInfo(BaseModel):
    symbol: str = Field(..., description="6位A股代码")
    code: Optional[str] = Field(None, description="兼容字段，优先使用symbol")
    name: str
    market: str = "A股"
    industry: Optional[str] = None
    sector: Optional[str] = None
    market_cap: Optional[float] = None
    price: Optional[float] = None
    change_percent: Optional[float] = None


class SingleAnalysisRequest(BaseModel):
    symbol: Optional[str] = Field(None, description="6位A股代码")
    stock_code: Optional[str] = Field(None, description="兼容字段，优先使用symbol")
    parameters: Optional[AnalysisParameters] = None

    def get_symbol(self) -> str:
        return self.symbol or self.stock_code or ""


class AnalysisTaskResponse(BaseModel):
    task_id: str
    symbol: str
    stock_code: Optional[str] = None
    stock_name: Optional[str] = None
    status: AnalysisStatus
    progress: int
    created_at: datetime
    started_at: Optional[datetime]
    completed_at: Optional[datetime]
    result: Optional[AnalysisResult]

    @field_serializer("created_at", "started_at", "completed_at")
    def serialize_datetime(self, value: Optional[datetime], _info) -> Optional[str]:
        return value.isoformat() if value else None


class AnalysisHistoryQuery(BaseModel):
    status: Optional[AnalysisStatus] = None
    start_date: Optional[datetime] = None
    end_date: Optional[datetime] = None
    symbol: Optional[str] = None
    stock_code: Optional[str] = None
    page: int = Field(default=1, ge=1)
    page_size: int = Field(default=20, ge=1, le=100)

    def get_symbol(self) -> Optional[str]:
        return self.symbol or self.stock_code
